services/spiget/spiget-rating.service.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
```

Comment:
You can simplify this to just have a single class definition (`SpigetRatings`). The only conditional part is really the value of the `message` being set in the `render` function. You could either leverage a class exporter function, or update the route to handle the two formats. The latter approach would have a route function like this:

```js
static get route() {
  return {
    base: 'spiget',
    pattern: ':format(rating|stars)/:resourceid',
  }
}
```

and then update the `handl` function to include the format parameter and pass it along as an argument to the render function:

```js
async handle({ format, resourceid }) {
  const { rating } = await this.fetch({ resourceid })
  return this.constructor.render({
    format,
    total: rating.count,
    average: rating.average,
  })
}
```

and then finally move the conditional rendering logic to your render function with something like:

```js
static render({ format, total, average }) {
  const message = format === 'stars'
    ? starRating(average)
    : `${average}/5 (${metric(total)})`
  return {
    message,
    color: floorCount(average, 2, 3, 4),
  }
}
```